src/driver.rs

```diff
@@ -410,6 +410,90 @@ impl DriverManager {
         Ok(Driver { c_driver })
     }
 
+    /// Get the Driver based on the file extension from filename
+    ///
+    /// Searches for the available extensions in the registered
+    /// drivers and returns the matches. The determined driver is
+    /// checked for writing capabilities as
+    /// [`Dataset::open`](Dataset::open) can already open datasets
+    /// with just path.
+    ///
+    /// See also: [`get_driver_by_name`](Self::get_driver_by_name)
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    /// use gdal::DriverManager;
+    /// # fn main() -> gdal::errors::Result<()> {
+    /// let compatible_drivers =
+    ///     DriverManager::get_drivers_for_filename("test.gpkg", true)
+    ///         .iter()
+    ///         .map(|d| d.short_name())
+    ///         .collect::<Vec<String>>();
+    /// println!("{:?}", compatible_drivers);
+    /// # Ok(())
+    /// # }
+    /// ```
+    /// ```text
+    /// ["GPKG"]
+    /// ```
+    pub fn get_drivers_for_filename(filename: &str, is_vector: bool) -> Vec<Driver> {
```

Comment:
Why is `is_vector` necessary? Is it possible to have multiple drivers that read the same  file extension but differentiate on vector vs. raster data? If so, is this something we need to impose on every user (IOW, could we let the caller deal with that if it's a rare edge case?)

I guess what I'm wondering here is if we should only be evaluating against `DMD_EXTENSION` and `DMD_EXTENSIONS`, and defer capability filtering to a different mechanism (e.g. extension method on `Driver`). The method name implies we're only looking at matching drivers against filename (extension).

Reply:
I don't know. In principle, the user might have all sorts of extra drivers installed. Matching the GDAL logic seemed like a safe approach.

We could ask Even, I guess.

Reply:
The logic I went by is this: there can be multiple drivers for the same extension, and some could support raster some vector and some both. And this function is (related with other comments) made to know the driver in advance while creating a file as if we have a file already, then we can call `Dataset::open` without driver information. But we cannot create and save data without deciding the driver first. And if we're writing a data, we'd want the list of drivers that supports that type of data, hence that bool. 

It was also in the function Inicola sent me as an example, so it's my understanding of that, not my innovation.